File: Problems/No.2367_Number_of_Arithmetic_Triplets/number_of_arithmetic_triplets.cpp
class Solution {
public:
    int arithmeticTriplets(vector<int>& nums, int diff) {
        
        // parameters
        int count = 0;
        int num_size = nums.size();

        // start computing
        for (int i=0; i<num_size; i++) {

            int j = i + 1;
            int k = num_size - 1;
            while (j < k) {
                
                int ij_diff = nums[j] - nums[i];
                int jk_diff = nums[k] - nums[j];

                if (ij_diff == diff) {

                    if (jk_diff == diff) {

                        count++;
                        break;

                    }
                    else if (jk_diff < diff) break;
                    else k--;

                }
                else if (ij_diff > diff) break;
                else j++;

            }

        }

        return count;

    }
};

File: Problems/No.2367_Number_of_Arithmetic_Triplets/number_of_arithmetic_triplets.py
class Solution(object):
    def arithmeticTriplets(self, nums, diff):
        """
        :type nums: List[int]
        :type diff: int
        :rtype: int
        """

        # parameters
        count = 0
        num_size = len(nums)

        # start computing
        for i in range(num_size):
            j = i + 1
            k = num_size - 1
            while (j < k):
                ij_diff = nums[j] - nums[i]
                jk_diff = nums[k] - nums[j]
                if (ij_diff == diff):
                    if (jk_diff == diff):
                        count += 1
                        break
                    elif (jk_diff < diff):
                        break
                    else:
                        k -= 1
                elif (ij_diff > diff):
                    break
                else:
                    j += 1
        
        return count
